env/env.cc

```diff
@@ -10,17 +10,58 @@
 #include "rocksdb/env.h"
 
 #include <thread>
+
 #include "env/composite_env_wrapper.h"
 #include "logging/env_logger.h"
 #include "memory/arena.h"
 #include "options/db_options.h"
 #include "port/port.h"
 #include "rocksdb/options.h"
+#include "rocksdb/system_clock.h"
 #include "rocksdb/utilities/object_registry.h"
 #include "util/autovector.h"
 
 namespace ROCKSDB_NAMESPACE {
 namespace {
+class LegacySystemClock : public SystemClock {
+ private:
+  Env* target_;
+
+ public:
+  explicit LegacySystemClock(Env* t) : target_(t) {}
+  const char* Name() const override { return "Legacy System Clock"; }
+
+  // Returns the number of micro-seconds since some fixed point in time.
+  // It is often used as system time such as in GenericRateLimiter
+  // and other places so a port needs to return system time in order to work.
+  uint64_t NowMicros() override { return target_->NowMicros(); }
+
+  // Returns the number of nano-seconds since some fixed point in time. Only
+  // useful for computing deltas of time in one run.
+  // Default implementation simply relies on NowMicros.
+  // In platform-specific implementations, NowNanos() should return time points
+  // that are MONOTONIC.
```

Comment:
NowMicros does not currently guarantee a monotonic clock.  It uses gettimeofday instead of the steady clocks.  Under Windows, it might use system_clock (not steady_clock) or GetSystemTimePreciseAsFileTime (which may be monotonic)

It *probably should* be monotonic (I did not write that comment but copied it from env.h) but would prefer to address that issue in a future PR.